triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerImpl.scala

```diff
@@ -24,28 +26,45 @@ import com.daml.ledger.client.configuration.{
   LedgerClientConfiguration,
   LedgerIdRequirement
 }
+import com.daml.jwt.domain.Jwt
+
+import java.util.UUID
+import java.time.Duration
 
 object TriggerRunnerImpl {
   case class Config(
+      server: ActorRef[Server.Message],
+      triggerId: UUID, // trigger id
+      jwt: Jwt, // trigger token
       compiledPackages: CompiledPackages,
       trigger: Trigger,
       ledgerConfig: LedgerConfig,
       maxInboundMessageSize: Int,
+      maxFailureNumberOfRetries: Int,
+      failureRetryTimeRange: Duration,
       party: Party,
   )
 
   import TriggerRunner.{Message, Stop}
-  final case class Failed(error: Throwable) extends Message
-  final case class QueryACSFailed(cause: Throwable) extends Message
-  final case class QueriedACS(runner: Runner, acs: Seq[CreatedEvent], offset: LedgerOffset)
+  final private case class Failed(error: Throwable) extends Message
+  final private case class QueryACSFailed(cause: Throwable) extends Message
+  final private case class QueriedACS(runner: Runner, acs: Seq[CreatedEvent], offset: LedgerOffset)
       extends Message
+  import Server.{
+    TriggerStarting,
+    TriggerStarted,
+    TriggerInitializationFailure,
+    TriggerRuntimeFailure
+  }
 
-  def apply(config: Config)(
+  def apply(parent: ActorRef[Message], config: Config)(
       implicit esf: ExecutionSequencerFactory,
       mat: Materializer): Behavior[Message] =
     Behaviors.setup { ctx =>
-      implicit val ec: ExecutionContext = ctx.executionContext
       val name = ctx.self.path.name
+      implicit val ec: ExecutionContext = ctx.executionContext
+      // Report to the server that this trigger is starting.
+      config.server ! TriggerStarting(config.triggerId, config.jwt, parent)
```

Comment:
Why do we need to report that we are starting? The server doesn't do anything in that case, but do you expect that to change?

Reply:
yes, i plan to introduce a table tracking changes so that we can report on the status of a trigger